### PR TITLE
Fixes SI-6758: force LazyAnnnotationInfo for DefDef and TypeDef

### PR DESCRIPTION
**Pull request by hubertp**
Looks like the change in 25ecde037f22ff no longer forced
lazy annotations for some of the cases.
Also removed forcing for PackageDef annotations as we currently
don't support them.

Review by @lrytz

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -2218,7 +2218,7 @@ trait Typers extends Modes with Adaptations with Tags {
       val tparams1 = tdef.tparams mapConserve typedTypeDef
       val typedMods = typedModifiers(tdef.mods)
       // complete lazy annotations
-      val annots = tdef.symbol.annotations
+      tdef.symbol.annotations.map(_.atp)
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Calling the exact same closure-creating map call on the same memberof Symbol  in four different places is more than sufficient cause to write a method.

Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Especially when it's purely for the side effect.

**Comment by hubertp** [comment]
Comment addressed, also fixed a failure due to a recent merge of #1691.
PLS REBUILD ALL

**Comment by lrytz** [comment]
lo-go-to-me

**Pull request merged**